dashboard/client/index.js

```diff
@@ -1,137 +1,146 @@
+  instance.render = data => {
+    if (data) {
+      nv.addGraph(() => {
+        const chart = nv.models.lineWithFocusChart()
 
-    const data = instance.parseEsData(chartData);
+        const tickMultiFormat = d3.time.format.multi([
+          ['%-d %b %y', d => d.getDate()],
+          ['%b %-d', d => d.getMonth()],
+          ['%Y', () => true]
+        ])
 
-    const dataForNV = instance.parseDataForNV(data);
-
-    if (dataForNV.data) {
-      nv.addGraph(function() {
-        var chart = nv.models.lineWithFocusChart();
-
-        var tickMultiFormat = d3.time.format.multi([
-          ["%b %-d", function(d) { return d.getDate() != 1; }], // not the first of the month
-          ["%b %-d", function(d) { return d.getMonth(); }], // not Jan 1st
-          ["%Y", function() { return true; }]
-        ]);
-
-        chart.interpolate("basis")
+        chart.interpolate('basis')
 
         chart.xAxis
-          .tickFormat((d) => tickMultiFormat(new Date(d)));
+          .tickFormat(d => tickMultiFormat(new Date(d)))
+          .axisLabel('Time')
 
         chart.x2Axis
-          .tickFormat((d) => tickMultiFormat(new Date(d)));
+          .tickFormat(d => tickMultiFormat(new Date(d)))
+          .axisLabel('Overview chart')
 
         chart.yAxis
-        .tickFormat(d3.format(',.2'));
+          .tickFormat(d3.format(',.2'))
+          .axisLabel('Calls')
 
         chart.y2Axis
-        .tickFormat(d3.format(',.2'));
+          .tickFormat(d3.format(',.2'))
+
+        chart.margin({ left: 70, right: 40 })
 
         d3.select('#chart svg')
           .attr('height', 500)
-          .datum(dataForNV.data)
-          .transition().duration(500)
-          .call(chart);
-
-        nv.utils.windowResize(chart.update);
+          .datum(data)
+          .transition()
+          .duration(500)
+          .call(chart)
 
-        return chart;
-      });
+        nv.utils.windowResize(chart.update)
 
+        return chart
+      })
     }
   }
-});
 
-Template.dashboard.onRendered(function(){
-  const instance = this;
+  instance.init = opts => {
+    instance.getAggrData(opts)
+      .then(items => {
+        instance.render(items)
+      })
+      .catch(err => console.error(err))
+  }
 
-  const opts = {
-    size: 10000,
-    body: {
-      sort: [
-        {
-          date: {
-            order: 'desc',
-          },
-        },
-      ],
-    },
-  };
+  instance.updateQuery = () => {
+    const from = FlowRouter.getQueryParam('from') || moment().subtract(1, 'month').format('YYYY-MM-DD')
+    const to = FlowRouter.getQueryParam('to') || moment().format('YYYY-MM-DD')
+    const granularity = FlowRouter.getQueryParam('granularity') || 'day'
+    const type = FlowRouter.getQueryParam('type') || ''
+
+    // For metics collection
+    // const a = moment(from, 'YYYY-MM-DD')
+    // const b = moment(to, 'YYYY-MM-DD')
+    // console.log(`${b.diff(a, 'days')} days shown.`)
+
+    const q = instance.opts.body.query.bool.must
```

Comment:
What does `q` refer to? Lets give this variable a semantic name, so we can [avoid mental mapping](https://github.com/ryanmcdermott/clean-code-javascript#avoid-mental-mapping).